src/variables.jl

```diff
@@ -995,3 +995,104 @@ end
 
 value(::_MA.Zero) = 0.0
 value(x::Number) = x
+
+function _info_from_variable(v::VariableRef)
+    has_lb = has_lower_bound(v)
+    lb = has_lb ? lower_bound(v) : -Inf
+    has_ub = has_upper_bound(v)
+    ub = has_ub ? upper_bound(v) : Inf
+    has_fix = is_fixed(v)
+    fixed_value = has_fix ? fix_value(v) : NaN
+    start_or_nothing = start_value(v)
+    has_start = !(start_or_nothing isa Nothing)
+    start = has_start ? start_or_nothing : NaN
+    has_start = start !== Nothing
+    binary = is_binary(v)
+    integer = is_integer(v)
+    return VariableInfo(has_lb, lb, has_ub, ub, has_fix, fixed_value,
+                        has_start, start, binary, integer)
+end
+
+"""
+    relax_integrality(model::Model)
+
+Modifies `model` to "relax" all binary and integrality constraints on
+variables. Specifically,
+
+- Binary constraints are deleted, and variable bounds are tightened if
+  necessary to ensure the variable is constrained to the interval ``[0, 1]``.
+- Integrality constraints are deleted without modifying variable bounds.
+- Semi-continuous and semi-integer constraints are ignored (left in place).
```

Comment:
No error for semi-continuous, semi-integer, SOS constraints then?

Reply:
I added an error for semi-continuous and semi-integer because we might want to add support for those in the future (and removing an error isn't a breaking change). I think SOS, indicator and other constraints are clearly out of scope (the name of the function is `relax_integrality`) and should be ignored. We can't possibly enumerate all types of constraints that might cause the problem to have discrete aspects.